src/_pytest/pathlib.py

```diff
@@ -305,3 +305,8 @@ def fnmatch_ex(pattern, path):
     else:
         name = six.text_type(path)
     return fnmatch.fnmatch(name, pattern)
+
+
+def parts(s):
```

Comment:
please take a look if a `Path.parents` or `Path.parts` works out here as well as this looks like a  nice potential match

Reply:
Using `p = Path(s); list(p.parents) + [p]` makes it slower.
This would only make sense if it would be a `Path` already I think.

Reply:
No surprise there really. Path.parents is a pretty complex chain of stuff that is hard to follow. Would be amazing if it wasn't slower than the simple string operations in this function.